alembic/versions/9008aa5065b4_add_taxprofiler_analysis_option.py

```diff
@@ -0,0 +1,47 @@
+"""Add taxprofiler analysis option
+
+Revision ID: 9008aa5065b4
+Revises: df1b3dd317d0
+Create Date: 2023-04-19 13:46:29.137152
+
+"""
+from alembic import op
+from sqlalchemy.dialects import mysql
+
+
+# revision identifiers, used by Alembic.
+revision = "9008aa5065b4"
+down_revision = "df1b3dd317d0"
+branch_labels = None
+depends_on = None
+
+old_options = (
+    "balsamic",
+    "balsamic-pon",
+    "balsamic-qc",
+    "balsamic-umi",
+    "demultiplex",
+    "fastq",
+    "fluffy",
+    "microsalt",
+    "mip-dna",
+    "mip-rna",
+    "rnafusion",
+    "rsync",
+    "sars-cov-2",
+    "spring",
+)
+new_options = sorted(old_options + ("taxprofiler",))
+
+old_enum = mysql.ENUM(*old_options)
+new_enum = mysql.ENUM(*new_options)
+
+
+def upgrade():
+    op.alter_column("family", "data_analysis", type_=new_enum)
+    op.alter_column("analysis", "pipeline", type_=new_enum)
+
+
+def downgrade():
```

Comment:
If we downgrade but have analyses/families with the value `taxprofiler`  it will throw an error since it is not an allowed value anymore. Add a query here where you fetch all famillies and analyses with the value and change the fields.

Reply:
Something like this, and the same for Family
```suggestion
def downgrade():
    bind = op.get_bind()
    session = sa.orm.Session(bind=bind)
    for analysis in session.query(Analysis).filter(Analysis.pipeline=="taxprofiler"):
        analysis.pipeline = "something_else"

    session.commit()

```

Reply:
Thanks for the review @Vince-janv. I will connect to the database but I am wondering where do the `Analysis `come from? Is it from one of the database tables?

Reply:
I define these two classes with the link to the tables in the db and their primary keys.

```
class Family(Base):
    __tablename__ = "data_analysis"
    id = sa.Column(sa.types.Integer, primary_key=True)
class Analysis(Base):
    __tablename__ = "pipeline"
    id = sa.Column(sa.types.Integer, primary_key=True)
```

Regarding your suggestion about `def_downgrade`, would that make sense to add:

`analysis.pipeline = "fastq"`

I assume that for the data where only fastq files are delivered and no analysis is being performed, the pipeline column in the database is called fastq, right?

![Screenshot 2023-04-27 at 17 36 32](https://user-images.githubusercontent.com/91951607/234913338-9e403749-c8d7-49e9-896a-c727e1d8b545.png)

Reply:
I will resolve this as well. @Vince-janv and I tested that alembic upgrade and downgrade works as expected.